lib/portage/dbapi/_MergeProcess.py

```diff
@@ -224,9 +225,10 @@ def _target(
         unmerge,
         vardb,
     ):
-        """
-        TODO: Make all arguments picklable for the multiprocessing spawn start method.
-        """
+        if QueryCommand._db is None and not hasattr(portage, "db"):
+            # Create portage.db for AbstractEbuildProcess/EbuildIpcDaemon
+            # when not using the multiprocessing fork start method.
+            portage._reset_legacy_globals()
```

Comment:
Maybe it would be nicer to copy down `QueryCommand._db` from the parent. Using `portage._reset_legacy_globals()` could possibly lead to there being duplicate dbapi instances in memory, which would not be nice.

Reply:
Let's try copy it down, I think? I wasn't sure when I was looking earlier as was on mobile and I couldn't look at _reset_legacy_globals and friends.

Reply:
There was some difficulty pickling all of `QueryCommand._db`, so I pruned it down to just the required vartree and porttree keys, and that worked:
```python
# Since QueryCommand._db is not entirely picklable, select only
# tree types that QueryCommand specifically requires.
child_db = {}
for root in QueryCommand._db:
	child_db[root] = {}
	for tree_type in ("vartree", "porttree"):
		child_db[root][tree_type] = QueryCommand._db[root][tree_type]
```